src/librustdoc/html/static/js/search-core.js

```diff
@@ -0,0 +1,3322 @@
+// ignore-tidy-filelength
+/* global searchState, exports */
+
+"use strict";
+
+// polyfill
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSpliced
+if (!Array.prototype.toSpliced) {
+    // Can't use arrow functions, because we want `this`
+    Array.prototype.toSpliced = function() {
+        const me = this.slice();
+        Array.prototype.splice.apply(me, arguments);
+        return me;
+    };
+}
+
+// This mapping table should match the discriminants of
+// `rustdoc::formats::item_type::ItemType` type in Rust.
+const itemTypes = [
```

Comment:
To prevent putting all these variables in the global scope, you need to wrap all this code in an anonymous function.